src/Swift.Bindings/src/Emitter/StringEmitter/Handler/MethodHandler.cs

```diff
@@ -624,130 +691,167 @@ private void EmitSwiftError(IndentedTextWriter writer)
-            var unsafeKeyword = _requiresIndirectResult || _env.MethodDecl.IsGeneric ? "unsafe " : "";
+            var unsafeKeyword = _requiresIndirectResult || _requiresSwiftAsync || _env.MethodDecl.IsGeneric ? "unsafe " : "";
 
-            writer.WriteLine($"public {staticKeyword}{unsafeKeyword}{_wrapperSignature.ReturnType} {_env.MethodDecl.Name}{genericParams}({_wrapperSignature.ParametersString()})");
+            var returnType = _wrapperSignature.ReturnType;
+            if (_requiresSwiftAsync)
+            {
+                returnType = $"Task{(_env.MethodDecl.CSSignature.First().SwiftTypeSpec.IsEmptyTuple ? "" : $"<{_wrapperSignature.ReturnType}>")}";
+            }
+
+            csWriter.WriteLine($"public {staticKeyword}{unsafeKeyword}{returnType} {_env.MethodDecl.Name}{genericParams}({_wrapperSignature.ParametersString()})");
         }
 
         /// <summary>
         /// Emits the finally block.
         /// </summary>
-        private void EmitFinally(IndentedTextWriter writer)
+        private void EmitFinally(CSharpWriter csWriter, SwiftWriter swiftWriter)
         {
-            writer.WriteLine("finally");
-            EmitBodyStart(writer);
+            csWriter.WriteLine("finally");
+            EmitBodyStart(csWriter, swiftWriter);
 
             foreach (var argument in _env.MethodDecl.CSSignature.Skip(1).Where(a => a.IsGeneric))
             {
                 var (_, _, payloadName) = _env.GenericTypeMapping[argument.SwiftTypeSpec.ToString()];
-                writer.WriteLine($"NativeMemory.Free((void*){payloadName});");
+                csWriter.WriteLine($"NativeMemory.Free((void*){payloadName});");
             }
 
-            EmitBodyEnd(writer);
+            EmitBodyEnd(csWriter, swiftWriter);
         }
 
         /// <summary>
         /// Emits the try block start.
         /// </summary>
-        private void EmitTryBlockStart(IndentedTextWriter writer)
+        private void EmitTryBlockStart(CSharpWriter csWriter, SwiftWriter swiftWriter)
         {
-            writer.WriteLine("try");
-            EmitBodyStart(writer);
+            csWriter.WriteLine("try");
+            EmitBodyStart(csWriter, swiftWriter);
         }
 
         /// <summary>
         /// Emits the try block end.
         /// </summary>
-        private void EmitTryBlockEnd(IndentedTextWriter writer)
+        private void EmitTryBlockEnd(CSharpWriter csWriter, SwiftWriter swiftWriter)
         {
-            EmitBodyEnd(writer);
+            EmitBodyEnd(csWriter, swiftWriter);
         }
 
         /// <summary>
         /// Emits the body start.
         /// </summary>
-        /// <param name="writer">The IndentedTextWriter instance.</param>
-        private void EmitBodyStart(IndentedTextWriter writer)
+        /// <param name="csWriter">The IndentedTextWriter instance.</param>
+        private void EmitBodyStart(CSharpWriter csWriter, SwiftWriter swiftWriter)
```

Comment:
Those methods don't really do anything with `swiftWriter`, maybe we should move them to `CSharpWriter` altogether

Reply:
Yes, we should remove it when not used. Alternatively, create a single `StringWriter` type with two `IdentedTextWriter`'s.

Reply:
Passed as an argument only when used.